src/components/notes/sharing.jsx

```diff
@@ -7,27 +7,35 @@ import styles from 'components/notes/sharing.styl'
 
 export default function SharingWidget(props) {
   const client = useClient()
-
   const file = useFileWithPath({ cozyClient: client, file: props.file })
-  const noteId = file && file.id
-
   const [showModal, setShowModal] = useState(false)
   const onClick = useCallback(() => setShowModal(!showModal), [showModal])
   const onClose = useCallback(() => setShowModal(false), [])
 
+  if (!file) return null
+
+  const {
+    id: noteId,
+    attributes: { name }
+  } = file
+
   return (
-    (file && (
-      <>
-        <ShareButton
-          theme="primary"
-          docId={noteId}
-          onClick={onClick}
-          extension="narrow"
-          className={styles['sharing-button']}
+    <>
+      <ShareButton
+        theme="primary"
+        docId={noteId}
+        onClick={onClick}
+        extension="narrow"
+        className={styles['sharing-button']}
+      />
+      {showModal && (
+        <ShareModal
+          document={{ ...file, name }}
```

Comment:
cozy-client does not give us a filename here but it is absolutely needed in order to have a working sharing by email. Should we see why cozy-client does not give a filename at note root or is the current workaround good enough?

Reply:
https://github.com/cozy/cozy-client/blob/master/packages/cozy-client/src/models/file.js#L125-L131

https://github.com/cozy/cozy-client/blob/master/packages/cozy-client/src/models/file.js#L111-L115


I don't see anything that remove the name attribute?